util/cache/disk.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package cache
+
+import (
+	"encoding/json"
+	"os"
+	"time"
+)
+
+// Disk is a cache that stores data in a file on-disk. It also supports
```

Comment:
Is there a concern about lateral movement, that anything which can write to this file can fool us into using a key which will match a hostile control server?

Reply:
Yeah, right now this is using the memory-only cache for that reason, along with the more general question of cache expiry/etc. If we were going to switch to the Disk cache, we'd probably want to add some mitigations, like trying to load the key from the server on startup in all cases, ignoring the presence of the cache and the timestamp in it, and only fallback to the cached value if that fails.

Reply:
Removed the disk cache entirely in the other PR